r/R/dataset-write.R

```diff
@@ -53,6 +53,34 @@
 #' - `codec`: A [Codec] which will be used to compress body buffers of written
 #'   files. Default (NULL) will not compress body buffers.
 #' @return The input `dataset`, invisibly
+#' @examples
+#' dout1 <- paste0(tempdir(), "/direct")
+#' dout2 <- paste0(tempdir(), "/group_by")
+#'
+#' # partitioning ----
+#' write_dataset(mtcars, dout1, "feather",
+#'  partitioning = "cyl",
+#'  basename_template = "{i}.feather",
+#'  hive_style = F)
```

Comment:
I wonder if we should leave off the `basename_template` and `hive_style` arguments here for clarity — the main focus here is on how the partitioning is defined and those might be more of a distraction than a help.